store/tikv/2pc.go

```diff
@@ -775,6 +776,20 @@ func (c *twoPhaseCommitter) execute(ctx context.Context) (err error) {
 	prewriteBo := NewBackofferWithVars(ctx, PrewriteMaxBackoff, c.txn.vars)
 	start := time.Now()
 	err = c.prewriteMutations(prewriteBo, c.mutations)
+
+	if err != nil {
+		// TODO: Now we return an undetermined error as long as one of the prewrite
+		// RPCs fails. However, if there are multiple errors and some of the errors
+		// are not RPC failures, we can return the actual error instead of undetermined.
```

Comment:
Can we make it completely done in this PR?

Reply:
I don't have a clear idea how to do...
I think it's not easy to know whether it's not an RPC error. For example, we will always retry after an RPC error. In the retry, the RPC error will disappear, and a region error will be finally returned back. Then it's possible that we get a backoff error of region miss but the real cause is RPC failure.
Any good idea?

Reply:
Can't we do it in a similar way to the original 2PC commit: https://github.com/pingcap/tidb/blob/master/store/tikv/commit.go#L51-L87

Reply:
It's a bit different... There is no concurrency when committing the primary so it can unset the undeterminedErr, but here we cannot because we might unset the error set by other concurrent goroutines.
During the retry, the original request can be re-grouped. So it's also not easy to manage the exact undeterminedErr set before.